landlab/components/species_evolution/species_evolver.py

```diff
@@ -0,0 +1,673 @@
+    taxa of this model can inhabit.
+
+    >>> def zone_func(grid):
+    ...     return grid.at_node['topographic__elevation'] < 100
+
+    Instantiate ZoneController with the grid and zone function. The initial
+    zones are created at controller instantiation. In this example, one zone is
+    created because all nodes of the zone mask are adjacent to each other.
+
+    >>> zc = ZoneController(mg, zone_func)
+    >>> len(zc.zones) == 1
+    True
+
+    Additional examples of controller usage are provided in ZoneController
+    documentation.
+
+    The ``mask`` of the zone is True where the conditions of the zone function
+    are met. All nodes of the grid are included because the elevation of each
+    node is below 100.
+
+    >>> zc.zones[0].mask
+    array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
+            True,  True,  True,  True,  True,  True,  True,  True,  True,
+            True,  True,  True], dtype=bool)
+
+    Populate a taxon to the zone. The attribute, ``taxa_data_frame`` indicates
+    only the one taxon exists because we populated each zone with one taxon,
+    and only one zone exists.
+
+    >>> taxon = zc.populate_zones_uniformly(1)
+    >>> se.track_taxa(taxon)
+    >>> se.taxa_data_frame
+         appeared  latest_time  extant
+    uid
+    0           0            0    True
+
+    Force a change in the zone mask to demonstrate component functionality.
+    Here we begin a new time step where topography is uplifted by 200 that
+    forms a ridge trending north-south in the center of the grid.
+
+    >>> z[[3, 10, 17]] = 200
+    >>> z.reshape(mg.shape)
+    array([[   1.,    1.,    1.,  200.,    1.,    1.,    1.],
+           [   1.,    1.,    1.,  200.,    1.,    1.,    1.],
+           [   1.,    1.,    1.,  200.,    1.,    1.,    1.]])
+
+    The current elevation, the elevation following uplift, is represented here.
+
+    - - - ^ - - -       elevation:  - 1
+    - - - ^ - - -                   ^ 200
+    - - - ^ - - -
+
+    The updated zone mask is below.
+
+    . . . x . . .       key:    . node in zone mask
+    . . . x . . .               x node outside of zone mask
+    . . . x . . .
+
+    Run a step of both the ZoneController and SpeciesEvolver. Two zones exist
+    following this time step.
+
+    >>> delta_time = 1000
+    >>> zc.run_one_step(delta_time)
+    >>> se.run_one_step(delta_time)
+    >>> len(zc.zones) == 2
+    True
+
+    A additional zone was created because the zone mask was not continuous.
+
+    . . . ^ * * *       key:    . a zone
+    . . . ^ * * *               * another zone
+    . . . ^ * * *               ^ mountain range
+
+    The split of the initial zone triggered speciation. Taxon 0 became extinct
+    as it speciated to child taxa, 1 and 2.
+
+    >>> se.taxa_data_frame
+         appeared  latest_time  extant
+    uid
+    0           0         1000   False
+    1        1000         1000    True
+    2        1000         1000    True
```

Comment:
Why does taxon 0 die and taxa 1 and 2 evolve? I thought what happens to taxa when a zone gets split up is under the control of the user. Is this just default behavior? If yes, maybe best to say?

Reply:
What happens to taxa is programmed into the Taxon classes. Its up to the Taxon programmer, which is in this case me, because I made ZoneTaxon demoed here.

The tree directly below this section shows taxon 0 doesn't die, but evolves into taxa 1 and 2 because, as you pointed out, the zone splits.

I added this short summary under the tree:

> The split of the initial zone into two zones at time 1000 triggered taxon 0
    to evolve into two child taxon objects. Taxon 1 occupies a zone on one side
    of the mountain range, and taxon 2 occupies a zone on the other side.

Reply:
OK, so maybe you could just include the text you just gave me - i.e. the Taxon programmer can change these details. That was not obvious to me here, but I thought it was the case.

Reply:
Added text.